File: source/zombie/asset/AssetType.java
// Decompiled on Sat Jan 17 08:24:00 EST 2026 with Zomboid Decompiler v0.2.3 using Vineflower.
package zombie.asset;

import java.util.zip.CRC32;

public final class AssetType {
    public static final AssetType INVALID_ASSET_TYPE = new AssetType("");
    public long type;

    public AssetType(String typeName) {
        CRC32 crc32 = new CRC32();
        crc32.update(typeName.getBytes());
        this.type = crc32.getValue();
    }
}

File: ProjectZomboidSwitch/src/zombie/asset/AssetType.cpp
#include <string>
#include "zombie/asset/AssetType.h"

namespace zombie {
namespace asset {

AssetType::AssetType(const std::string& var1)
    : type(0)
{
    // CRC32 calculation placeholder; set type based on var1 string
    // Example: type = crc32(var1.c_str(), var1.size());
}
}

} // namespace asset
} // namespace zombie
